app/js/components/homePage/manageApps/manageApps.jsx

```diff
@@ -8,21 +8,48 @@
  */
 import React from 'react';
 
+import { ApiHelper } from '../../../helpers/apiHelper';
+
 export default class ManageApps extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      appList: []
+    };
+
+    this.openPage = this.openPage.bind(this);
+  }
+  componentWillMount() {
+    const apiHelper = new ApiHelper(null);
+    apiHelper.get('/owa/applist').then(response => {
+      if (response.error) throw response.error.message;
+      this.setState((prevState, props) => {
+        return {
+          appList: response
+        };
+      });
+    });
+  }
+
+  openPage(app) {
+    return location.href = `/${location.href.split('/')[3]}/owa/${app.folderName}/${app.launch_path}`;
+  }
+
   render() {
+    const { appList } = this.state;
     return (
       <div>
         <h3 id="manageApps">Manage Add Ons</h3>
         <div className="manage-app-buttons">
           <button type="button" className="button-add-module ">
-            <span className="glyphicon glyphicon-plus span-icon"></span>  Add/Upgrade Add ons
+            <span className="glyphicon glyphicon-plus span-icon"/>  Add/Upgrade Add ons
           </button> 
           <button type="button" className="button-search-module">
-            <span className="glyphicon glyphicon-search span-icon search-icon"></span>Search Add ons
+            <span className="glyphicon glyphicon-search span-icon search-icon"/>Search Add ons
           </button> 
         </div>
-        <div className="manage-app-table col-sm-10">
-          <table className="table table-bordered">
+        <div className="manage-app-table col-sm-12">
```

Comment:
@kingisaac95 Can we move the `listing Addon table` to a new component, e.g `ListAddon.jsx`

Reply:
I've done that. I have `AddonList.jsx` that renders the table body.
Thanks

Reply:
I've done that. I have AddonList.jsx that renders the table body.
Thanks @andela-jomadoye